pkg/common/utils/utils.go

```diff
@@ -265,3 +270,70 @@ func QueryAllVolumesForCluster(ctx context.Context, m cnsvolume.Manager, cluster
 	}
 	return queryAllResult, nil
 }
+
+func GetVirtualMachineAllApiVersions(ctx context.Context, vmKey types.NamespacedName,
+	vmOperatorClient client.Client) (*vmoperatorv1alpha1.VirtualMachine, *vmoperatorv1alpha2.VirtualMachine,
+	*vmoperatorv1alpha3.VirtualMachine, error) {
+	log := logger.GetLogger(ctx)
+	vmV1alpha1 := &vmoperatorv1alpha1.VirtualMachine{}
+	vmV1alpha2 := &vmoperatorv1alpha2.VirtualMachine{}
+	vmV1alpha3 := &vmoperatorv1alpha3.VirtualMachine{}
+	var err error
+	log.Info("get machine with vm-oeprator api version v1alpha3")
+	if err = vmOperatorClient.Get(ctx, vmKey, vmV1alpha3); err != nil {
```

Comment:
Is it possible to check specific error that type is not registered or something specific before falling back to prior version. all other errors should be returned.

Reply:
check added to confirm type is not registered.
made changes to return other errors and log  errors.